### PR TITLE
RDK-35329: animated splash screen support with APNG

### PR DESCRIPTION
**Pull request by madanagopalt**

### CODE REVIEW

**Comment by mhughesacn** [comment]
Hi madangolapt: The scan is complaining because there are good matches to pxCore.  If you have another original source in the two files below, please tell me.  Otherwise, you need to credit pxCore as follows:

1.
Code match in rdkshellimage.cpp, please insert the following before Line 704 (BlendOver,loadAPng)

```
Based on pxCore, Copyright 2005-2018 John Robinson
Licensed under the Apache License, Version 2.0
```

2.
Code match in apngutils.cpp, please insert before line 173 (blit):

```
pxCore Copyright 2005-2018 John Robinson
Licensed under the Apache License, Version 2.0

```
3.
In NOTICE, please append:

```
pxCore Copyright 2005-2018 John Robinson
Licensed under the Apache License, Version 2.0

```

Thank you - Martin Hughes, RDK CMF Compliance Team

**Review by mfiess — CHANGES_REQUESTED** [review]
Please make the requested changes.

File: rdkshellimage.cpp [hunk 1]
```diff
@@ -635,6 +701,230 @@ namespace RdkShell
         return true;
     }
 
+    void Image::BlendOver(unsigned char **rows_destination, unsigned char **rows_source, unsigned int x, unsigned int y, unsigned int width, unsigned int height)
```
Comment (mfiess) [inline rdkshellimage.cpp:704]: Please ensure the method names are camel case starting with a lower case letter to keep consistency with the other RDKShell APIs.

File: rdkshellimage.h [hunk 2]
```diff
@@ -45,7 +46,9 @@ namespace RdkShell
             bool loadJpeg(std::string fileName, unsigned char *&image, int32_t &width, int32_t &height);
             bool loadPng(std::string fileName, unsigned char *&image, int32_t &width, int32_t &height);
             bool loadBmp(std::string fileName, unsigned char *&image, int32_t &width, int32_t &height);
+            bool loadAPng(std::string fileName, unsigned char *&image, int32_t &width, int32_t &height);
             bool loadPngFromData(const char* imageData, int32_t imageSize, unsigned char *&image, int32_t &width, int32_t &height);
+            void BlendOver(unsigned char **rows_destination, unsigned char **rows_source, unsigned int x, unsigned int y, unsigned int width, unsigned int height);
```
Comment (mfiess) [inline rdkshellimage.h:51]: Please ensure the method names are camel case starting with a lower case letter to keep consistency with the other RDKShell APIs.

**Review by mfiess — CHANGES_REQUESTED** [review]

**Pull request closed without merging**